### DISCUSSION TITLE
When or can geometry shaders be implemented into MoltenVK?

### DISCUSSION THREAD

**Discussion by loyahdev** [KhronosGroup/MoltenVK#1785] [General]
Like for macOS there’s metal 3 with mesh shaders but when will it be implemented for moltenvk?

**Comment by billhollings** [comment]
There is some PoC effort currently underway using Mesh shaders to provide Geometry shaders.

We're expecting initial availability to be sometime in Q1/23.

**Reply by blasphemousjohn** [reply]
That's it, no going back now; I'm going to hold you to this date Bill ;)
 /s

**Reply by siegmund-heiss-ich** [reply]
Is there an ETA for Q1/23 to be finished? 😅

**Reply by billhollings** [reply]
@blasphemousjohn @siegmund-heiss-ich 

Geometry Shaders are under development in PR #1815, but resourcing is limited, so it's proceeding slowly.

**Reply by sofakng** [reply]
Glad to see this is still being worked on and I wanted to say thanks for all of your hard work!  (@billhollings)

Is this issue the best place to follow any updates?

**Reply by loyahdev** [reply]
We got any updates @billhollings or just places we should go to and find more info?

**Reply by billhollings** [reply]
The current plan for geometry shaders is to use Apple's new Shader Converter tech. As for more info, we've got this identified in th [MoltenVK roadmap](https://github.com/KhronosGroup/MoltenVK/issues/1975).

**Comment by gdkchan** [comment]
Will the geometry shader implementation support subgroup operations? I'm currently working on a emulator where some games require this feature.
